numba_cuda/numba/cuda/dispatcher.py

```diff
@@ -700,6 +694,16 @@ def __call__(self, *args):
             args, self.griddim, self.blockdim, self.stream, self.sharedmem
         )
 
+    def __getstate__(self):
+        state = self.__dict__.copy()
+        state["stream"] = int(state["stream"].handle)
+        return state
+
+    def __setstate__(self, state):
+        handle = state.pop("stream")
+        self.__dict__.update(state)
+        self.stream = driver._to_core_stream(handle)
```

Comment:
How is the current `Stream` class handled for pickling? Capturing the `handle` which is a ptr in a pickle feels wrong.

Reply:
Yes, pickling a pointer is indeed weird, however, that is the status quo, as `Stream` uses the default pickling implementation and its fields are `handle` (stream handle), and `external` (bool).

Reply:
I was chatting with @cpcloud about this and yeah, serializing a stream class handle feels weird to me.  What does that even mean to deserialize a Stream handle? I think we are probably getting lucky that the majority of handles are 0 representing the default stream.

Reply:
I will create an issue to track this.